lib/Amazon/S3/Thin.pm

```diff
@@ -148,6 +148,41 @@ sub list_objects {
     return $response;
 }
 
+sub delete_multiple_objects {
+    my ($self, $bucket, @keys) = @_;
+
+    my $content = _build_xml_for_delete( @keys );
+
+    require Digest::MD5;
+    my $request = $self->_compose_request(
+        'POST',
+        "$bucket/?delete",
+        {
+            'Content-MD5'    => Digest::MD5::md5_base64($content) . '==',
+            'Content-Length' => length $content,
+        },
+        $content
+    );
+    my $response = $self->ua->request($request);
+    return $response;
+}
+
+sub _build_xml_for_delete {
+    my (@keys) = @_;
```

Comment:
what happens if you pass 1001 keys ? ( I don't know :) )
maybe we should warn ( or die or carp or croak) in such a case.